### PR TITLE
chore: support for selecting llm with different namespace

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
chore: support for selecting llm with different namespace

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer
```yaml
# rating
apiVersion: core.kubebb.k8s.com.cn/v1alpha1
kind: Rating
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"core.kubebb.k8s.com.cn/v1alpha1","kind":"Rating","metadata":{"annotations":{},"name":"two-dimension-rating","namespace":"kubebb-system"},"spec":{"componentName":"kubebb.kubebb-core","evaluator":{"llm":{"kind":"LLM","name":"zhipuai","namespace":"kubebb-system"}},"pipelineParams":[{"dimension":"reliability","params":[{"name":"URL","value":{"arrayVal":[],"objectVal":{},"stringVal":"https://github.com/kubebb/components/releases/download/kubebb-core-v0.1.10/kubebb-core-v0.1.10.tgz","type":"string"}},{"name":"COMPONENT_NAME","value":{"stringVal":"kubebb-core","type":"string"}},{"name":"VERSION","value":{"stringVal":"v0.1.10","type":"string"}},{"name":"REPOSITORY_NAME","value":{"stringVal":"repository-kubebb","type":"string"}}],"pipelineName":"kubebb-core-reliability"},{"dimension":"security","params":[{"name":"URL","value":{"arrayVal":[],"objectVal":{},"stringVal":"https://github.com/kubebb/components/releases/download/kubebb-core-v0.1.10/kubebb-core-v0.1.10.tgz","type":"string"}},{"name":"COMPONENT_NAME","value":{"stringVal":"kubebb-core","type":"string"}},{"name":"VERSION","value":{"stringVal":"v0.1.10","type":"string"}},{"name":"REPOSITORY_NAME","value":{"stringVal":"kubebb-core","type":"string"}}],"pipelineName":"kubebb-core-security"}]}}
  creationTimestamp: "2024-01-09T15:27:41Z"
  generation: 2
  labels:
    rating.component: kubebb.kubebb-core
    rating.repository: kubebb
  name: two-dimension-rating
  namespace: kubebb-system
  resourceVersion: "215673777"
  uid: 3d86168f-e868-417b-98a1-76793a840de3
spec:
  componentName: kubebb.kubebb-core
  evaluator:
    llm:
      kind: LLM
      name: zhipuai
      namespace: kubebb-system
  pipelineParams:
  - dimension: reliability
    params:
    - name: URL
      value:
        stringVal: https://github.com/kubebb/components/releases/download/kubebb-core-v0.1.10/kubebb-core-v0.1.10.tgz
        type: string
    - name: COMPONENT_NAME
      value:
        stringVal: kubebb-core
        type: string
    - name: VERSION
      value:
        stringVal: v0.1.10
        type: string
    - name: REPOSITORY_NAME
      value:
        stringVal: repository-kubebb
        type: string
    pipelineName: kubebb-core-reliability
  - dimension: security
    params:
    - name: URL
      value:
        stringVal: https://github.com/kubebb/components/releases/download/kubebb-core-v0.1.10/kubebb-core-v0.1.10.tgz
        type: string
    - name: COMPONENT_NAME
      value:
        stringVal: kubebb-core
        type: string
    - name: VERSION
      value:
        stringVal: v0.1.10
        type: string
    - name: REPOSITORY_NAME
      value:
        stringVal: kubebb-core
        type: string
    pipelineName: kubebb-core-security
status:
  conditions:
  - lastTransitionTime: "2024-01-09T15:27:59Z"
    message: Evaluation succeeded
    reason: EvaluationSucceeded
    status: "True"
    type: Ready
  evaluations:
    reliability:
      conditions:
      - lastTransitionTime: "2024-01-09T15:26:20Z"
        message: Finished CallLLM
        reason: ReconcileSuccess
        status: "True"
        type: Done
      prompt: two-dimension-rating-reliability
    security:
      conditions:
      - lastTransitionTime: "2024-01-09T15:26:28Z"
        message: Finished CallLLM
        reason: ReconcileSuccess
        status: "True"
        type: Done
      prompt: two-dimension-rating-security
  pipelineRuns:
    reliability:
      conditions:
      - lastTransitionTime: "2024-01-09T15:27:48Z"
        message: 'Tasks Completed: 1 (Failed: 0, Cancelled 0), Skipped: 0'
        reason: Succeeded
        status: "True"
        type: Succeeded
      pipelineName: kubebb-core-reliability
      pipelinerunName: two-dimension-rating.reliability
      tasks:
      - conditions:
        - lastTransitionTime: "2024-01-09T15:27:48Z"
          message: All Steps have completed executing
          reason: Succeeded
          status: "True"
          type: Succeeded
        description: ""
        name: kubebb-core-helm-lint
        taskRunName: two-dimension-rating.reliability-kubebb-core-helm-lint
    security:
      conditions:
      - lastTransitionTime: "2024-01-09T15:27:51Z"
        message: 'Tasks Completed: 1 (Failed: 0, Cancelled 0), Skipped: 0'
        reason: Succeeded
        status: "True"
        type: Succeeded
      pipelineName: kubebb-core-security
      pipelinerunName: two-dimension-rating.security
      tasks:
      - conditions:
        - lastTransitionTime: "2024-01-09T15:27:51Z"
          message: All Steps have completed executing
          reason: Succeeded
          status: "True"
          type: Succeeded
        description: ""
        name: kubebb-core-rback
        taskRunName: two-dimension-rating.security-kubebb-core-rback
```

```shell
[root@kubebb-master rating]# kubectl get prompt -nkubebb-system
NAME                               AGE
two-dimension-rating-reliability   4m13s
two-dimension-rating-security      4m10s
[root@kubebb-master rating]# kubectl get po -nkubebb-system
NAME                                                         READY   STATUS      RESTARTS   AGE
chartmuseum-784f45dc4f-bqtdf                                 1/1     Running     0          82d
kubebb-core-677d66fcf4-sfmp5                                 1/1     Running     0          15m
two-dimension-rating.reliability-kubebb-core-helm-lint-pod   0/1     Completed   0          4m34s
two-dimension-rating.security-kubebb-core-rback-pod          0/1     Completed   0          4m34s
[root@kubebb-master rating]# kubectl get pipelinerun -nkubebb-system
NAME                               SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
two-dimension-rating.reliability   True        Succeeded   4m42s       4m35s
two-dimension-rating.security      True        Succeeded   4m42s       4m32s
```

86F1AC8A-E3BE-4E30-AC79-171E48AE9E7E

### CODE REVIEW

File: .github/workflows/example_test.yaml [hunk 1]
```diff
@@ -97,6 +97,10 @@ jobs:
         with:
           key: operator.image-${{ github.sha }}
           path: /tmp/operator.image.tar
+      - name: Set up Go
+         uses: actions/setup-go@v4
+         with:
+           go-version: "1.21"
```
Comment (Abirdcfly) [inline .github/workflows/example_test.yaml:103]: go-version-file: 'go.mod'

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -4,10 +4,18 @@ on:
...
+  GO_TAGS: ""
+
 jobs:
   verify-code:
     runs-on: ubuntu-latest
     steps:
+      - name: Set up Go
+         uses: actions/setup-go@v4
+         with:
+           go-version: "1.21"
```
Comment (Abirdcfly) [inline .github/workflows/verify.yml:18]: go-version-file: 'go.mod' and Change it after `actions/checkout`

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20 as builder
+FROM golang:1.21.6 as builder
```
Comment (Abirdcfly) [inline Dockerfile:1]: ARG GO_VER=1.21.6
FROM golang:${GO_VER} as builder

File: .github/workflows/example_test.yaml [hunk 4]
```diff
@@ -31,9 +32,9 @@ jobs:
         # see https://github.com/actions/runner/issues/662
         if: steps.cache.outputs.cache-hit != 'true'
       - name: Build operator image
-        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (Abirdcfly) [inline .github/workflows/example_test.yaml:34]: restore this line

**Review by Abirdcfly — APPROVED** [review]

**Review by bjwswang — APPROVED** [review]

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**